File: Tool/template/c++/Template0.cpp
#include "TemplateClass.h"

std::vector<TemplateClass*>* TemplateClass::m_Datas;

int TemplateClass::GetCount()
{
	if(m_Datas == NULL)
	{
		return 0;
	}
    return m_Datas->size();
}

std::vector<TemplateClass*>* TemplateClass::GetDatas()
{
    return m_Datas;
}



void TemplateClass::Load(BinData *bin)
{
ReadValueLoadContent	
}



void TemplateClass::Load() 
{
	BinData* bin = OpenBin("TemplateClass.bin");
	if (bin != NULL && bin->IsOpen())
	{
		try
		{
			int count = bin->ReadInt32();
			m_Datas = new std::vector<Test*>(count);
			for (int i = 0; i < count; ++i)
			{
				TemplateClass * data = new TemplateClass();
				data->Load(bin);
				m_Datas->push_back(data);
			}
		}
		catch (const std::exception& e)
		{

		}
		bin->Close();
		delete bin;
		bin = NULL;
	}
}

void TemplateClass::Reload()
{
    Unload();
    TemplateClass::Load();
}

void TemplateClass::Unload()
{
  	if (m_Datas == NULL)
	{
		return;
	}

	for (std::vector<Test*>::iterator it = m_Datas->begin(); it != m_Datas->end(); it++) {
		delete *it;
	}

	m_Datas->clear();
	delete m_Datas;
	m_Datas = NULL;
}

File: Tool/template/java/Template0.java
package TableConfig;


public class TemplateClass extends DataBase
{
	//region value
ValueContent
	//endregion
	//region load and get funtion
	private static TemplateClass[] m_Datas = null;
	public static TemplateClass[] GetDatas()
	{
		return m_Datas;
	}

	
	public static int GetCount()
	{
		if(m_Datas == null)
		{
			return 0;
		}
		return m_Datas.length;
	}

	private void Load(BinData bin)
	{
ReadValueLoadContent		
	}


	public static void Load()
	{
		BinData bin = OpenData("TemplateClass.bin");
		if (bin != null)
		{
			try
			{
				int count = bin.ReadInt32();
				m_Datas = new TemplateClass[count];
				for (int i = 0; i < count; ++i)
				{
					TemplateClass data = new TemplateClass();
					data.Load(bin);
					m_Datas[i] = data;
				}
			}
			catch(Exception e)
			{
				
			}
		}
	}

	public static void Reload()
	{
		Unload();
		Load();
	}


	public static void Unload()
	{
		boolean bGC = false;
		if(m_Datas != null)
		{
			m_Datas = null;
			bGC = true;
		}
		if(bGC)
		{
			System.gc();
		}
	}
	//endregion
}